### ISSUE TITLE
BigInteger.ToByteArray extension method collides with System.Numerics.BigInteger starting with .NET Standard 2.1

### ISSUE THREAD

**Issue by misha-fm** [Nethereum/Nethereum#671]
## Expected Behavior
There is no confusion between .ToByteArray methods when passing a single parameter.

## Current Behavior
With .NET Standard 2.0 the method BigInteger.ToByteArray(**false**) is called from Nethereum package and means a **big endian** notation.
With .NET Standard 2.1 the method BigInteger.ToByteArray(**false**) is called from MS package and means a **little endian** notation and signed value.

## Possible Solution
Either provide a new method name, or make the current one not intersecting with the MS implementation.

## Detailed Description
In .NET Standard 2.0 and earlier, the type System.Numerics.BigInteger had a single overload of .ToByteArray() method accepting no arguments (https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.tobytearray?view=netstandard-2.0):
`public byte[] ToByteArray ();`

Starting with .NET Standard 2.1, this type has received a second overload of this method that accepts two optional parameters (https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.tobytearray?view=netstandard-2.1): 
`public byte[] ToByteArray (bool isUnsigned = false, bool isBigEndian = false);`

Nethereum package has its own extension method with the same name and a single accepted argument:
https://github.com/Nethereum/Nethereum/blob/bba79dd97d29370353b908730f2cf59c591916c3/src/Nethereum.Hex/HexConvertors/Extensions/HexBigIntegerConvertorExtensions.cs#L9

So upgrade from .NET Standard 2.0 to .NET Standard 2.1 brings completely different behavior of the aforementioned method.

Moreover, the MS overloaded method now accepts a bool `isBigEndian = false` parameter, while in Nethereum the parameter has the opposite meaning: `bool littleEndian`

**Comment by juanfranblanco** [comment]
@misha-fm yes I know about this, Nethereum was first, so it is obviously pretty annoying what is happening, as first it will break everybody's code if I change the code, as users will end up using by mistake the .Net's one. 

The best solution is to use an alias, or use a full namespace.

**Comment by juanfranblanco** [comment]
Note by an alias, I mean wrapping it. Another option to don't break other peoples code, might be to add an extra method, although that may add more confusion.

**Comment by Stalesamy** [comment]
new System.Numerics.BigInteger(50).ToHexBigInteger()); 

getting an error for : the type 'BigInteger' exists in both 'System.Numerics, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' and 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

**Comment by juanfranblanco** [comment]
You should not be getting that error, what is your project layout?

**Comment by Stalesamy** [comment]
using System;
using System.Numerics;
using System.Threading.Tasks;
using Nethereum.Hex.HexTypes;
using Nethereum.Web3;
using WalletConnectSharp.Core;
using WalletConnectSharp.Core.Models;
using UnityEngine;

namespace WalletConnectSharp.Examples.Examples
{
    public class ConnectExample : IExample
    {
        public string Name
        {
            get
            {
                return "connect_example";
            }
        }

        public async Task Execute(string[] args)
        {
            var clientMeta = new ClientMeta()
            {
                Name = "WalletConnectSharp",
                Description = "An example that showcases how to use the WalletConnectSharp library",
                Icons = new []{ "https://app.warriders.com/favicon.ico" },
                URL = "https://app.warriders.com/"
            };

            var client = new WalletConnect(clientMeta);
            
            Debug.Log("Connect using the following URL");
            Debug.Log(client.URI);

            await client.Connect();
            
            Debug.Log("The account " + client.Accounts[0] + " has connected!");

            Debug.Log("Using RPC endpoint " + args[0] + " as the fallback RPC endpoint");
            var web3 = new Web3(client.CreateProvider(new Uri(args[0])));

            var firstAccount = client.Accounts[0];

            var secondAccount = args[1];
            
            Debug.Log("Sending test transactions from " + firstAccount + " to " + secondAccount);

            try
            {
                await web3.Eth.TransactionManager.SendTransactionAsync(firstAccount, secondAccount,
                    new System.Numerics.BigInteger(50).ToHexBigInteger()); 
            }
            catch(Exception e)
            {
                Debug.Log(e);
            }


            await client.Disconnect();
        }
    }
}

**Comment by Stalesamy** [comment]
sorry for the copy paste but couldn't get it to work

**Comment by juanfranblanco** [comment]
Sure what environment?

**Comment by Stalesamy** [comment]
unity3d

**Comment by Stalesamy** [comment]
using walletconnectsharp plugin https://github.com/WalletConnect/WalletConnectSharp/tree/feature/unity

**Comment by juanfranblanco** [comment]
Delete the extra System.Numerics you will find in the Assets folder.

**Comment by Stalesamy** [comment]
now I get: 'BigInteger' does not contain a definition for 'ToHexBigInteger' and no accessible extension method 'ToHexBigInteger' accepting a first argument of type 'BigInteger' could be found (are you missing a using directive or an assembly reference?)

**Comment by Stalesamy** [comment]
I removed the system.numerics dll

**Comment by juanfranblanco** [comment]
Have you added the Nethereum dlls? Also if you want to come to discord, because this is a completely different issue and is just support.

**Comment by Stalesamy** [comment]
will do, thanks
